test/test-fs.c

```diff
@@ -1580,6 +1580,19 @@ TEST_IMPL(fs_access) {
   ASSERT(access_cb_count == 1);
   access_cb_count = 0; /* reset for the next test */
 
+  /* File should grant read and write permission */
+  r = uv_fs_access(NULL, &req, "test_file", F_OK | R_OK | W_OK, NULL);
```

Comment:
My reading of https://linux.die.net/man/2/access and https://pubs.opengroup.org/onlinepubs/9699919799/functions/access.html suggests that the `F_OK | R_OK | W_OK` combination of flags is not valid (not just on z/OS) and not expected to work. @zsw007 Is there some context as to the reason this PR was created -- perhaps z/OS was doing the right thing all along?

Reply:
In Node.js, under the parallel/test-fs-access test case, `F_OK | R_OK | W_OK` is being tested for:

https://github.com/nodejs/node/blob/162cc4fa6303e3dd3a1f7ec041f9aa073a40d7d7/test/parallel/test-fs-access.js#L156

This Node.js test case currently passes on platforms other than z/OS, and the changes in this PR is needed for the test case to pass on z/OS as well. The extra test case that I added to this PR is based off that Node.js test case (which currently also passes on platforms other than z/OS).

But if you're saying that `F_OK | R_OK | W_OK` is not valid, then I'm wondering what's the reason for it being tested in Node.js?

Reply:
> In Node.js, under the parallel/test-fs-access tests, `F_OK | R_OK | W_OK` is being tested for on line 156:
> 
> https://github.com/nodejs/node/blob/162cc4fa6303e3dd3a1f7ec041f9aa073a40d7d7/test/parallel/test-fs-access.js#L156
> 
> This Node.js test case currently passes on platforms other than z/OS, and the changes in this PR is needed for the test case to pass on z/OS as well. The extra test case that I added to this PR is based off that Node.js test case (which currently also passes on platforms other than z/OS, and needs this PR to pass on z/OS).
> 
> But if you're saying that `F_OK | R_OK | W_OK` is not valid, then I'm wondering what's the reason for it being tested in Node.js?

The Node.js test appears to have been added in https://github.com/nodejs/node/pull/114 which was authored by... @cjihrig 😄.

Reply:
Past Colin comes back to haunt me again. I'll open a PR to fix the Node.js test.

Reply:
Ah ok, so if ORing F_OK is a mistake in that test case, then it looks like this PR isn't necessary and the test case can be fixed instead 🙂